src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,249 @@
+begin
+  rcases hf with ⟨f', hf₁, cf, hcf, lif, hf₂⟩,
+  rcases hg with ⟨g', hg₁, cg, hcg, lig, hg₂⟩,
+  exact ⟨g'.comp f', has_fderiv_at.comp x hg₁ hf₁, cg * cf, mul_ne_zero hcg hcf, lig.comp lif,
+    by { ext, simp [coe_comp' f' g', hf₂, hg₂, smul_smul cg cf] }⟩,
+end
+
+lemma const_smul {f : X → Y} {x : X} {c : ℝ} (hc : c ≠ 0) (hf : conformal_at f x) :
+  conformal_at (c • f) x :=
+let p := (conformal_at_const_smul hc $ f x).comp x hf in
+begin
+  have : c • f = (λ x', c • x') ∘ f := by ext; simp only [pi.smul_apply, function.comp_app],
+  rw this, exact p,
+end
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+lemma _root_.conformal_at_iff {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) : conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : E), ⟪f' u, f' v⟫ = (c : ℝ) * ⟪u, v⟫ :=
+begin
+  split,
+  { rintros ⟨f₁, hf, c₁, hc₁, li, hf₁⟩,
+    exact (hf.unique h) ▸ (is_conformal_map_iff f₁).mp ⟨c₁, hc₁, li, hf₁⟩, },
+  { rintros ⟨c, hc, huv⟩,
+    exact ⟨f', h, (is_conformal_map_iff f').mpr ⟨c, hc, huv⟩⟩, },
+end
+
+/-- The conformal factor of a conformal map at some point `x`. Some authors refer to this function
+    as the characteristic function of the conformal map. -/
+def conformal_factor_at {f : E → F} (x : E) {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) (H : conformal_at f x) : ℝ :=
+by choose c hc huv using (conformal_at_iff h).mp H; exact c
+
+/-- If a real differentiable map `f` is conformal at a point `x`,
+    then it preserves the angles at that point. -/
+lemma _root_.preserves_angle {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) (H : conformal_at f x) (u v : E) :
+  inner_product_geometry.angle (f' u) (f' v) = inner_product_geometry.angle u v :=
+let ⟨f₁, h₁, c₁, hc₁, li, h₂⟩ := H in
+begin
+  rw has_fderiv_at.unique h h₁, exact is_conformal_map.preserves_angle f₁ ⟨c₁, hc₁, li, h₂⟩ u v,
+end
+
+end conformal_at
+
+
+/-- A map `f` is conformal on a set `s` if it's conformal at every point in that set. -/
+def conformal_on (f : X → Y) (s : set X) :=
+∀ (x : X), x ∈ s → conformal_at f x
+
+namespace conformal_on
+
+lemma conformal_at {f : X → Y} {u : set X} (h : conformal_on f u) {x : X} (hx : x ∈ u) :
+conformal_at f x := h x hx
+
+lemma comp {f : X → Y} {g : Y → Z}
+  {u : set X} {v : set Y} (hf : conformal_on f u) (hg : conformal_on g v) :
+  conformal_on (g ∘ f) (u ∩ f⁻¹' v) :=
+λ x hx, (hg (f x) $ set.mem_preimage.mp hx.2).comp x (hf x hx.1)
+
+lemma congr {f : X → X} {g : X → X}
+  {u : set X} (hu : is_open u) (h : ∀ (x : X), x ∈ u → g x = f x) (hf : conformal_on f u) :
+  conformal_on g u := λ x hx, let ⟨f', h₁, c, hc, li, h₂⟩ := hf x hx in
+begin
+  have : has_fderiv_at g f' x :=
+    by apply h₁.congr_of_eventually_eq; rw filter.eventually_eq_iff_exists_mem;
+       exact ⟨u, hu.mem_nhds hx, h⟩,
+  exact ⟨f', this, c, hc, li, h₂⟩,
+end
```

Comment:
Some style points:
* Using `;` just to avoid `{}` is generally considered bad form - it's reserved for when you actually have multiple goals you want to work on at once
* Mixing `let` with `begin / end` is quite uncommon - you can use `obtain` inside the tactic block instead

```suggestion
  conformal_on g u :=
λ x hx, begin
  obtain ⟨f', h₁, c⟩ := hf x hx,
  refine ⟨f', h₁.congr_of_eventually_eq _, c⟩,
  rw filter.eventually_eq_iff_exists_mem,
  exact ⟨u, hu.mem_nhds hx, h⟩,
end
```
Also, once again there's no need to unfold `is_conformal_map` here.